hackathon_site/event/serializers.py

```diff
@@ -32,6 +32,35 @@ class Meta:
             "e_signature",
             "team",
         )
+        read_only_fields = ("id", "team", "acknowledge_rules", "e_signature")
+
+    def update(self, instance: Profile, validated_data):
+        if not validated_data:
+            raise serializers.ValidationError("no modifiable fields provided")
+        return super().update(instance, validated_data)
+
+
+class CurrentProfileSerializer(ProfileSerializer):
+    class Meta(ProfileSerializer.Meta):
+        read_only_fields = (
+            "id",
+            "team",
+            "id_provided",
+            "attended",
+        )
+
+    def update(self, instance: Profile, validated_data):
+        if not validated_data:
+            raise serializers.ValidationError("no modifiable fields provided")
+        acknowledge_rules = validated_data.pop("acknowledge_rules", False)
+        e_signature = validated_data.pop("e_signature", None)
+
+        if not instance.acknowledge_rules and acknowledge_rules:
+            instance.acknowledge_rules = acknowledge_rules
+        if not instance.e_signature and e_signature:
+            instance.e_signature = e_signature
+
+        return serializers.ModelSerializer.update(self, instance, validated_data)
```

Comment:
had to use the original update method because calling super() would call the ProfileSerializer update method and since we pop off the acknowledge_rules and e_signature, it would always throw a validation error

Reply:
Good catch! Could you add in a comment explaining that?